### PR TITLE
AzureBlob work

### PR DESCRIPTION
**Pull request by clguimanMSFT**
1. Download the blob in 4MB chunks #70
2. Add user agent

### CODE REVIEW

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb [hunk 1]
```diff
@@ -109,26 +109,35 @@ class LogStash::Inputs::LogstashInputAzureblob < LogStash::Inputs::Base
...
   MAX = 2 ** ([42].pack('i').size * 16 -2 ) -1
 
+  # 4MB
+  MAX_BUFFER_SIZE = 4 * 1024 * 1024 
+
+
   public
   def register
+    user_agent = "logstash-input-azureblob"
+    user_agent << "/" << Gem.latest_spec_for("logstash-input-azureblob").version.to_s
```
Comment (brahmnes) [inline Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: Make sure there is an easy way to query this across all versions, for example, something like prefix(user_agent) == "logstash-input-azureblob"

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb [hunk 2]
```diff
@@ -159,7 +168,15 @@ def process(queue)
             start_index = 0 if start_index < 0
           end
 
-          blob, content = @azure_blob.get_blob(@container, blob_name, {:start_range => start_index} )
+          blob_size = blob.properties[:content_length]
+          end_index = start_index + MAX_BUFFER_SIZE - 1
+          @sleep_interval = 0
```
Comment (brahmnes) [inline Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: An cleaner approach may be, have process() return a bool which represent "if I am still in the middle of a blob", and the caller checks and decides to sleep or not based on that. That way, you don't need this extra member called @sleep_internal and setting it to 0 or @interval.

File: Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb (hunk 1 above)
Comment (clguimanMSFT) [reply Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: I don't understand. The current user agent is "logstash-input-azureblob/0.9.11"

Comment (brahmnes) [reply Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb]: Imagine you have a database storing these data with rows storing them as
logstash-input-azureblob/0.9.8
logstash-input-azureblob/0.9.9
logstash-input-azureblob/0.9.10
...

The question is, can you write a query that returns count for all logstash version, irrespective of the specific plugin versions.

**Comment by clguimanMSFT** [comment]
This implementation will break JSON parsing. I'll need to also parse json fragments (#85) before I can merge this PR

**Comment by xiaomi7732** [comment]
Generally, I would suggest we split these 2 changes to 2 PRs, they are too different. The agent string a lightweight change as long as we agreed on how to build the string, that can't break too much stuff; 

The huge file issue, on the other than, is an interesting issue, it is important and could impact core functionalities. At the same time, there are interesting edge cases too, for example, what will happen if we page loading a json and the file is appended during the reading.

On the side, there are several other directions that might bring us benefits. For example, does the storage SDK supports streaming loading a blob - that we could totally avoid having the whole file loaded. Or, can we make use of the block info for azure blobs to load the file block by block.

**Comment by clguimanMSFT** [comment]
I've moved the user agent work in a separate PR: #101 
I'll close this one and open a new PR once the partial json problem is fixed

**Pull request closed without merging**